### PR TITLE
Backported code from the current master

### PR DESCRIPTION
**Pull request by mchf**
https://bugzilla.suse.com/show_bug.cgi?id=1028081

### CODE REVIEW

File: library/network/src/modules/NetworkInterfaces.rb [hunk 1]
```diff
@@ -681,6 +683,20 @@ def ConcealSecrets(devs)
       deep_copy(out)
     end
 
+    # Get current sysconfig configured interfaces
+    #
+    # @param devregex [String] regex to filter by
+    # @return [Array] of ifcfg names
+    def get_devices(devregex = "[~]")
```
Comment (jreidinger) [inline library/network/src/modules/NetworkInterfaces.rb]: why parameter is not regexp and instead only string that is later interpreted as regexp?

Comment (jreidinger) [inline library/network/src/modules/NetworkInterfaces.rb]: also why regexp is not just `~` ?

File: library/network/test/network_interfaces_helpers_test.rb [hunk 2]
```diff
@@ -71,5 +71,47 @@ module Yast
...
+        subject.main
+        allow(subject).to receive(:Read).and_return(true)
+        allow(Yast::SCR).to receive(:Dir).with(Yast::Path.new(".network.section")).and_return(devices)
+      end
+
+      it "returns an array of configured interfaces filtered by regexp" do
+        expect(subject.get_devices("1")).not_to include "em1", "eth1", "br1"
+      end
+
+      it "filters with <[~]> by default" do
```
Comment (jreidinger) [inline library/network/test/network_interfaces_helpers_test.rb]: maybe i woulld name it `filters devices with "~" by default`

File: library/network/test/network_interfaces_helpers_test.rb [hunk 3]
```diff
@@ -71,5 +71,47 @@ module Yast
...
+        expect(subject.get_devices).not_to include "tr~"
+      end
+
+      it "returns an empty array with <.> argument" do
+        expect(subject.get_devices(".")).to eql []
+      end
+
+      it "returns all devices filtering with <''>" do
```
Comment (jreidinger) [inline library/network/test/network_interfaces_helpers_test.rb]: `with empty string`

File: library/network/test/network_interfaces_helpers_test.rb [hunk 4]
```diff
@@ -71,5 +71,47 @@ module Yast
...
+        expect(subject.get_devices("")).to eql devices
+      end
+
+      it "does not crash with exception" do
```
Comment (jreidinger) [inline library/network/test/network_interfaces_helpers_test.rb]: quite strong statement and you are never sure that it do not return expection. E.g. low memory or signal can get almost in every line of code.

File: library/network/test/network_interfaces_helpers_test.rb [hunk 5]
```diff
@@ -71,5 +71,47 @@ module Yast
...
+        expect { subject.get_devices }.not_to raise_error
+      end
+
+      it "doesn't carry empty strings" do
```
Comment (jreidinger) [inline library/network/test/network_interfaces_helpers_test.rb]: `filters out empty strings`

File: library/network/src/modules/NetworkInterfaces.rb (hunk 1 above)
Comment (mchf) [inline library/network/src/modules/NetworkInterfaces.rb]: I don't know, ask @teclator ;-)

**Comment by mchf** [comment]
@jreidinger Are you fine with creating a PR with modifications based on your comments against master? This should be pure backport of code which already (more less) is in master.

**Comment by jreidinger** [comment]
@mchf yes, I am fine.

File: library/network/src/modules/NetworkInterfaces.rb (hunk 1 above)
Comment (teclator) [inline library/network/src/modules/NetworkInterfaces.rb]: You can improve it 👍 . It was one of my first cleanups and just use same match that was used with `Builtins.regexpmatch`

https://github.com/yast/yast-yast2/pull/433/files#diff-36f13a6d2bafc7ad7bcfa6f8a452e81dL704

PR: https://github.com/yast/yast-yast2/pull/433

**Comment by teclator** [comment]
@mchf @jreidinger, I'm fine with just merge this PR with the backported code but the improvements or modifications maybe would be great to do directly in SP1 and then merge into SP2 and master

File: library/network/src/modules/NetworkInterfaces.rb (hunk 1 above)
Comment (mchf) [inline library/network/src/modules/NetworkInterfaces.rb]: @teclator I know ;-) I'll do so for master not maintenance branch.

File: .travis.yml [hunk 6]
```diff
@@ -5,7 +5,7 @@ before_install:
     # disable rvm, use system Ruby
     - rvm reset
     - wget https://raw.githubusercontent.com/yast/yast-devtools/master/travis-tools/travis_setup.sh
-    - sh ./travis_setup.sh -p "rake yast2-core yast2-devtools yast2-testsuite yast2-ruby-bindings yast2 yast2-pkg-bindings" -g "rspec:3.3.0 yast-rake gettext simplecov coveralls rubocop:0.29.1"
+    - sh ./travis_setup.sh -p "rake yast2-core yast2-devtools yast2-testsuite yast2-ruby-bindings yast2 yast2-pkg-bindings" -g "rspec:3.3.0 yast-rake gettext simplecov:0.10.0 coveralls rubocop:0.29.1"
```
Comment (jreidinger) [inline .travis.yml]: better fix is to completelly remove simplecov and it pick right version

Comment (mchf) [inline .travis.yml]: ok

**Pull request merged**